internal/service/appstream/user.go

```diff
@@ -0,0 +1,209 @@
+		ReadWithoutTimeout:   resourceUserRead,
+		DeleteWithoutTimeout: resourceUserDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"authentication_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(appstream.AuthenticationType_Values(), false),
+			},
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"first_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(0, 2048),
+			},
+			"last_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(0, 2048),
+			},
+			"message_action": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(appstream.MessageAction_Values(), false),
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"user_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+		},
+	}
+}
+
+func resourceUserCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).AppStreamConn
+
+	userName := d.Get("user_name").(string)
+	authType := d.Get("authentication_type").(string)
+
+	input := &appstream.CreateUserInput{
+		AuthenticationType: aws.String(authType),
+		UserName:           aws.String(userName),
+	}
+
+	if v, ok := d.GetOk("first_name"); ok {
+		input.FirstName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("last_name"); ok {
+		input.LastName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("message_action"); ok {
+		input.MessageAction = aws.String(v.(string))
+	}
+
+	var err error
+	var _ *appstream.CreateUserOutput
+	err = resource.RetryContext(ctx, fleetOperationTimeout, func() *resource.RetryError {
+		_, err = conn.CreateUserWithContext(ctx, input)
+		if err != nil {
+			if tfawserr.ErrCodeEquals(err, appstream.ErrCodeResourceNotFoundException) {
+				return resource.RetryableError(err)
+			}
+
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if tfresource.TimedOut(err) {
+		_, err = conn.CreateUserWithContext(ctx, input)
+	}
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating Appstream User (%s): %w", d.Id(), err))
+	}
+
+	if _, err = waitUserAvailable(ctx, conn, userName, authType); err != nil {
+		return diag.FromErr(fmt.Errorf("error waiting for Appstream User (%s) to be available: %w", d.Id(), err))
+	}
```

Comment:
Is there a delay in user creation? If not, this isn't needed

Reply:
Yes, it seems there's a delay because without the wait function, it will panic a nil error during the read function